sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -139,6 +141,11 @@ object Literal {
    * Constructs a Literal from a String
    */
   def fromString(str: String, dataType: DataType): Literal = {
```

Comment:
wait actually, is it ever properly called somewhere previously? If not, we could just remove it. Since all classes in catalyst are considered an internal API, we could just remove if that's not called.

Reply:
Is there an explicit plan to use this somewhere? If so, that's okay.

Otherwise, let's remove. In general, we shouldn't keep the codes for internal purpose that are not being called. Removed codes remaining in the commit and other branches - we can always revive them when it's actually used or in order to deduplicate.

Reply:
I think we can remove it now. It was there for `TreeNode.fromJSON`, which has been removed long time ago.

Reply:
ok. Here is the PR to remove the methods: https://github.com/apache/spark/pull/23603